api/server/apps_get.go

```diff
@@ -18,7 +18,7 @@ func (s *Server) handleAppGet(c *gin.Context) {
 		return
 	}
 
-	app, err = s.datastore.GetAppByID(ctx, c.MustGet(api.AppID).(string))
+	app, err = s.datastore.GetAppByName(ctx, app.Name)
```

Comment:
the previous code looks correct here. we need to use app ids in all handlers

Reply:
Disagree with this. We can use `GetAppByID` here only in case if his handler covered by new middleware, the downside is that we would need to query 2 times of the same app object. So that's why this handler remains as it was before app ID.

Reply:
we are already querying 2 times for the same app object here, no? this still suffers the fate that there is a race in the delete-recreate case that returns the wrong app here if we use name. right now I see:

* we get the app before calling the handler to translate to id, store on context
* the handler extracts the app from 1)
* the handler calls the middleware with that app
* we look up the app by name (which could be a different app)
* the handler calls the middleware with a potentially different app

so this seems wrong in a couple of different ways. we want to let them modify the app before calling get app but when we get that app we want them to get the same app, not a potentially different one (there's a race if we get by name).